.github/workflows/microbenchmark.yml

```
@@ -0,0 +1,53 @@
+name: microbenchmark
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - 'feature/**'
+    paths-ignore:
+      - '**.md'
+      - '**.asciidoc'
+
+# limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+env:
+  GO_VERSION: 1.15.10
+
+jobs:
+  microbenchmark:
+    runs-on: ubuntu-latest
+    # wait up to 1 hour
+    timeout-minutes: 60
+    steps:
+      - id: buildkite
+        name: Run buildkite pipeline
+        uses: elastic/apm-pipeline-library/.github/actions/buildkite@current
+        with:
+          vaultUrl: ${{ secrets.VAULT_ADDR }}
+          vaultRoleId: ${{ secrets.VAULT_ROLE_ID }}
+          vaultSecretId: ${{ secrets.VAULT_SECRET_ID }}
+          pipeline: apm-agent-microbenchmark
+          waitFor: true
+          printBuildLogs: true
+          buildEnvVars: |
+            script=scripts/jenkins/bench.sh
+            repo=apm-agent-go
+            sha=feature/microbenchmark
+            GO_VERSION=${{ env.GO_VERSION }}
+            GO111MODULE=on
+            BUILD_TIMEOUT=40
```

To be used by the Buildkite pipeline... need to find out how to override the default timeout of 240 minutes, since `timeout-minutes: 60` does not have any precedence unfortunately.

---

See https://github.com/golang/go/issues/31481
```suggestion
            GOFLAGS=-pkgmodrw
```